test/e2e/user_management_test.go

```diff
@@ -42,7 +45,99 @@ func (s *userManagementTestSuite) TearDownTest() {
 	s.testCtx.Cleanup()
 }
 
+func (s *userManagementTestSuite) TestUserDeactivation() {
+	s.setApprovalPolicyConfig("automatic")
+
+	userSignup, mur := s.createAndCheckUserSignup(true, "iris-at-redhat-com", "iris@redhat.com", approvedByAdmin()...)
+
+	// Deactivate the user
+	userSignup.Spec.Deactivated = true
+	err := s.awaitility.Client.Update(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' set to deactivated", userSignup.Name)
+
+	err = s.hostAwait.WaitUntilMasterUserRecordDeleted(mur.Name)
+	require.NoError(s.T(), err)
+}
+
+func (s *userManagementTestSuite) TestUserBanning() {
+
+	s.T().Run("user banning", func(t *testing.T) {
```

Comment:
could you please also add a case when:
1. user was provisioned
2. user got banned
3. the ban was removed
4. user is provisioned again

Reply:
Done